### ISSUE TITLE
stability: pq: already committed in 1->3 upreplication test

### ISSUE THREAD

**Issue by cuongdo** [cockroachdb/cockroach#13879]
The 1->3 allocator test (10 GB of data, photos+block_writer load) failed with [this error](https://teamcity.cockroachdb.com/viewLog.html?buildId=166817&buildTypeId=Cockroach_AllocatorTest1to310g&tab=buildLog#_focus=1778):

```
[02:39:01][TestUpreplicate_1To3Small] allocator_test.go:202: pq: already committed
```

Prior to the issue, there are many slow (> 500 ms) `applyRaftCommand` calls in the [node.1/cockroach.stderr log](https://teamcity.cockroachdb.com/repository/download/Cockroach_AllocatorTest1to310g/166817:id/node.1/cockroach.stderr) and [node.2/cockroach.stderr log](https://teamcity.cockroachdb.com/repository/download/Cockroach_AllocatorTest1to310g/166817:id/node.2/cockroach.stderr), some of which last 25-30 seconds.

[The node.0/cockroach.stderr log](https://teamcity.cockroachdb.com/repository/download/Cockroach_AllocatorTest1to310g/166817:id/node.0/cockroach.stderr) looks much quieter but has several of these log messages:

```
W170302 07:33:19.935357 69 vendor/github.com/coreos/etcd/raft/raft.go:793  [n1,s1,r158/1:/Table/51/1/49{08467…-42903…},@c4207bf500] 1 stepped down to follower since quorum is not active
```

followed by many of these messages:

```
E170302 07:33:51.546791 223800 storage/replica_range_lease.go:194  [n1,s1,r223/1:/Table/51/1/8{890508…-925628…},@c4201b8380] heartbeat failed on epoch increment
W170302 07:33:51.546922 223813 storage/node_liveness.go:293  [n1,s1,r236/1:/Table/51/1/15{08328…-45226…},@c420239500] slow heartbeat took 2.4s
```

**Comment by cuongdo** [comment]
Assigning to @petermattis for triage

**Comment by cuongdo** [comment]
Another failure:

https://teamcity.cockroachdb.com/viewLog.html?buildId=167471&buildTypeId=Cockroach_AllocatorTest1to310g

**Comment by tamird** [comment]
Another failure:

https://teamcity.cockroachdb.com/viewLog.html?buildId=179900&buildTypeId=Cockroach_AllocatorTest1to310g

**Comment by spencerkimball** [comment]
This was almost certainly fixed by #14789, by relocating the error check in `queryTxnStatus` for whether the txn is already committed or aborted to only apply to the pusher's txn. Obviously it's not an error if the pushee's txn is already committed or aborted. You would have expected to see this only in situations where the system was really lagging, which jibes with these bug report.

@cuongdo please re-open and comment if you've seen this since #14789 landed.
